src/gt4py/analysis/passes.py

```diff
@@ -756,6 +757,16 @@ def apply(self, transform_data: TransformData):
             else:
                 self.iir.fields[name] = decl
 
+        # Emit warning if no api symbol (field or parameter) is used
+        if all(
+            not (symbol.in_use and symbol.is_api) for name, symbol in self.data.symbols.items()
+        ):
+            loc = self.data.definition_ir.loc
+            warnings.warn(
+                f"Stencil without effective computation specified at line {loc.line} (column {loc.column}).",
```

Comment:
the line numbers in loc will be relative to the function, so they are probably useless. I'm not sure if the word `effective` is the right choice. `"The specified computation has no effect."` maybe?

Reply:
If the name of the stencil would be given the line numbers were much more useful. Something for #122 I guess.